src/charm.py

```diff
@@ -86,14 +87,18 @@ def __init__(self, *args):
         # TODO:
         #  ingress route provisioning a separate TCP ingress for each receiver if GRPC doesn't work directly
 
-        self._ingress = IngressPerAppRequirer(
-            self, port=self.tempo.tempo_server_port, strip_prefix=True
-        )
-
+        self._ingress = TraefikRouteRequirer(self, self.model.get_relation("ingress"), "ingress")  # type: ignore
         self._tracing = TracingEndpointProvider(
-            self, host=self.tempo.host, external_url=self._ingress.url
+            self, host=self.tempo.host, external_url=self._ingress.external_host
         )
 
+        self.framework.observe(self.on["ingress"].relation_joined, self._configure_ingress)
+        self.framework.observe(self.on.leader_elected, self._configure_ingress)
+        self.framework.observe(self.on.config_changed, self._configure_ingress)
+        self.framework.observe(self._ingress.on.ready, self._on_ingress_ready)  # pyright: ignore
+        # TODO there's no revoked ingress action with traefik_route?
+        # self.framework.observe(self._ingress.on.revoked, self._on_ingress_revoked)
```

Comment:
```suggestion
       self.framework.observe(self._ingress.on.broken, self._on_ingress_broken)
```

Reply:
This is a bit confusing as I tried that and it looks like there's only `ready` for `TraefikRouteRequirer`? https://github.com/canonical/traefik-k8s-operator/blob/main/lib/charms/traefik_route_k8s/v0/traefik_route.py#L119

Reply:
Created an issue in traefik: https://github.com/canonical/traefik-k8s-operator/issues/345

IMO Grafana doesn't handle it either at the moment so this can be dealt with separately.